### ISSUE TITLE
Deprecate from numpy import *

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#13525]
An idea for `numpy/__init__.py`:
```python
if sys.version_info[:2] < (3, 7):
    __all__ += ['__don_t_do_this']

    def __getattr__(attr):
        if attr == '__don_t_do_this':
            warning.warn("Don't do this')
            return None
        raise AttributeError
```

**Comment by chhanganivarun** [comment]
But why?

**Comment by mhvk** [comment]
Until python itself starts to do this, I don't think we should enforce this.

**Comment by rgommers** [comment]
Agreed. This will breaks a huge amount of code, just to teach better habits. That's what docs are for. I suggest to close this.

Nice trick by the way:)

**Comment by seberg** [comment]
It was not the most serious suggestion to be begin with. But the trick may be useful if we deprecate other attributes. Since we can modify the deprecation warning in the case of `*` imports to not be a difficult to understand spam (especially difficult to understand when it comes to avoiding the warning).

**Comment by seberg** [comment]
Closing this, it wasn't very serious about actually deprecating `from numpy import *`, and I think we are now comfortable enough with `__getattr__` to not have an issue as a hint.

If we deprecate a name, it is also removed from `__all__`.
